nodejs/demos/trivial/adams.js

```diff
@@ -140,6 +119,7 @@ Adams.prototype.lookup = function() {
     }
     this.commit();
   }).bind(this));
```

Comment:
To save yourself from having to use .bind(this) everywhere, typically should just save a reference to this in a variable called `self` at the top of the function like such:

```js
Adams.prototype.lookup = function() {
  var self = this; // save reference to this out here
  
  // can continue to use this on this level (self is the same)
  this.datastore.datasets.lookup({ /* options */ }, function(err, result) {
    if(err) {
      // but in the anonymous function, use self because this now points to something else
      self.emit('error', err);
    }
  }); // no bind :) much cleaner
}